modules/codegen/src/main/scala/com/twilio/guardrail/generators/AkkaHttpClientGenerator.scala

```diff
@@ -45,13 +44,11 @@ object AkkaHttpClientGenerator {
         .fold(param"host: String")(v => param"host: String = ${Lit.String(v)}")
 
     def apply[T](term: ClientTerm[T]): Target[T] = term match {
-      case GenerateClientOperation(className, RouteMeta(pathStr, httpMethod, operation), tracing, protocolElems) =>
-        def generateUrlWithParams(path: String,
-                                  pathArgs: List[ScalaParameter],
-                                  qsArgs: List[ScalaParameter]): Target[Term] = {
+      case GenerateClientOperation(className, RouteMeta(pathStr, httpMethod, operation), tracing, protocolElems, generatorSettings) =>
+        implicit val gs = generatorSettings
```

Comment:
implicits need to have types defined

Reply:
I was trying to restructure things so `Target` would be a `ReaderT[F, GeneratorSettings, T]`, but I can't clear up the circular dependency; I'd much rather not have all these parameters everywhere

Reply:
I was able to come up with a decent middleground using `ReaderT`. If you wouldn't mind taking another look, I think this approach reads much cleaner.